deprecation.py

```diff
@@ -273,3 +285,56 @@ def test_inner(*args, **kwargs):
                      (method, str(warning.message))))
         return rv
     return test_inner
+
+
+def _deprecated_wrapper(func, warning_class, object_name, deprecated_in,
+                        removed_in, details):
+    def new(*args, **kwargs):
+        if warning_class:
+            _warn_user(
+                warning_class, object_name, deprecated_in,
+                removed_in, details
+            )
+        return func(*args, **kwargs)
+    return new
+
+
+class DeprecatedVariable(object):
+    def __new__(
+        cls, obj, object_name, deprecated_in=None, removed_in=None,
+        current_version=None, details=""
+    ):
+        warning_class = _get_warning_class(
+            deprecated_in, removed_in, current_version
+        )
+
+        class TemporaryClass(obj.__class__):
```

Comment:
This is going to blow up even for things like `SOME_THING_TO_BE_DEPRECATED = True`.